jsonschema/_validators.py

```diff
@@ -190,9 +190,14 @@ def enum(validator, enums, instance, schema):
 
 
 def ref(validator, ref, instance, schema):
-    with validator.resolver.resolving(ref) as resolved:
+    scope, resolved = validator.resolver.resolve(ref)
+    validator.resolver.push_scope(scope)
```

Comment:
Just curious -- did we benchmark whether the reason it's slow is a context manager vs specifically `functools.contextmanager`?

But yeah, sorry, you're right -- although you don't need to do that check here per se every time you hit this function, you can check to see if you have a new or old `RefResolver` in `__init__` for the validator and bind accordingly?